distribution/push_v2.go

```diff
@@ -61,15 +61,24 @@ type pushState struct {
 func (p *v2Pusher) Push(ctx context.Context) (err error) {
 	p.pushState.remoteLayers = make(map[layer.DiffID]distribution.Descriptor)
 
-	p.repo, p.pushState.confirmedV2, err = NewV2Repository(ctx, p.repoInfo, p.endpoint, p.config.MetaHeaders, p.config.AuthConfig, "push", "pull")
+	var transportOK bool
+	p.repo, p.pushState.confirmedV2, transportOK, err = NewV2Repository(ctx, p.repoInfo, p.endpoint, p.config.MetaHeaders, p.config.AuthConfig, "push", "pull")
 	if err != nil {
 		logrus.Debugf("Error getting v2 registry: %v", err)
-		return fallbackError{err: err, confirmedV2: p.pushState.confirmedV2}
+		return fallbackError{
+			err:         err,
+			confirmedV2: p.pushState.confirmedV2,
+			transportOK: transportOK,
+		}
 	}
 
 	if err = p.pushV2Repository(ctx); err != nil {
 		if continueOnError(err) {
-			return fallbackError{err: err, confirmedV2: p.pushState.confirmedV2}
+			return fallbackError{
+				err:         err,
+				confirmedV2: p.pushState.confirmedV2,
+				transportOK: true,
```

Comment:
> Should this use the variable value from above?

In the original version, `NewV2Repository` always returned `transportOK == true` on success, because completing this function successfully involves a successful HTTP "ping". Per your other comment, I've updated it to return `transportOK` as part of the error it returns. This should make it clearer that `transportOK` is only false in certain cases where `NewV2Repository` fails.